### PR TITLE
[AMORO-2517] Fix the initial sql script loading problem in Windows

### LINKED ISSUE #2517: [Bug]: Started AMS failed in Windows environment: Illegal char <:> at index 2: /xxx/derby/ams-derby-init.sql
### What happened?

When I tried to start AMS in my local Windows environment, it failed to load the derby initial script.

### Affects Versions

master

### What engines are you seeing the problem on?

AMS

### How to reproduce

Start AMS in Windows envrionment.

### Relevant log output

```shell
2024-01-30 11:03:40,070 ERROR [main] [com.netease.arctic.server.ArcticServiceContainer] [] - AMS encountered an unknown exception, will exist
java.lang.IllegalStateException: Create tables failed
	at com.netease.arctic.server.persistence.SqlSessionFactoryProvider.createTablesIfNeed(SqlSessionFactoryProvider.java:153) ~[classes/:?]
	at com.netease.arctic.server.persistence.SqlSessionFactoryProvider.init(SqlSessionFactoryProvider.java:115) ~[classes/:?]
	at com.netease.arctic.server.ArcticServiceContainer$ConfigurationHelper.initServiceConfig(ArcticServiceContainer.java:404) ~[classes/:?]
	at com.netease.arctic.server.ArcticServiceContainer$ConfigurationHelper.init(ArcticServiceContainer.java:384) ~[classes/:?]
	at com.netease.arctic.server.ArcticServiceContainer.initConfig(ArcticServiceContainer.java:195) ~[classes/:?]
	at com.netease.arctic.server.ArcticServiceContainer.<init>(ArcticServiceContainer.java:100) ~[classes/:?]
	at com.netease.arctic.server.ArcticServiceContainer.main(ArcticServiceContainer.java:106) [classes/:?]
Caused by: java.nio.file.InvalidPathException: Illegal char <:> at index 2: /D:/git/amoro/ams/server/target/classes/derby/ams-derby-init.sql
	at sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182) ~[?:1.8.0_202]
	at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153) ~[?:1.8.0_202]
	at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77) ~[?:1.8.0_202]
	at sun.nio.fs.WindowsPath.parse(WindowsPath.java:94) ~[?:1.8.0_202]
	at sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:255) ~[?:1.8.0_202]
	at java.nio.file.Paths.get(Paths.java:84) ~[?:1.8.0_202]
	at com.netease.arctic.server.persistence.SqlSessionFactoryProvider.createTablesIfNeed(SqlSessionFactoryProvider.java:148) ~[classes/:?]
	... 6 more
```

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by zhoujinsong**
## Why are the changes needed?

Close #2517.

## Brief change log

- Change loading initial SQL script codes from `Paths.get(String)` to `Paths.get(URI)`. 

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by zhoujinsong** [comment]
It works fine in my local Windows environment after the code change.
image

File: ams/server/src/main/java/com/netease/arctic/server/persistence/SqlSessionFactoryProvider.java [hunk 1]
```diff
@@ -42,6 +42,8 @@
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 
 import java.io.InputStreamReader;
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.net.URL;
```
Comment (XBaith) [inline ams/server/src/main/java/com/netease/arctic/server/persistence/SqlSessionFactoryProvider.java:47]: Repeatedly importing the identical class: `java.net.URI`?

Comment (zhoujinsong) [reply ams/server/src/main/java/com/netease/arctic/server/persistence/SqlSessionFactoryProvider.java:47]: Seems not to be. They are `java.net.URI` and `java.net.URL`.

Comment (XBaith) [reply ams/server/src/main/java/com/netease/arctic/server/persistence/SqlSessionFactoryProvider.java:47]: Sorry for the misreading.LGTM

**Review by XBaith — APPROVED** [review]
